### PR TITLE
engines/libpmem: minor fixes and code cleanup

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
- set file open/create mode always to RW,
- do not call drain on close, it's redundant,
- code and example cleanup.

### CODE REVIEW

File: engines/libpmem.c [hunk 1]
```diff
@@ -185,8 +175,9 @@ static enum fio_q_status fio_libpmem_queue(struct thread_data *td,
 	io_u->error = 0;
 
 	dprint(FD_IO, "DEBUG fio_libpmem_queue\n");
-	dprint(FD_IO,"td->o.odirect %d td->o.sync_io %d \n",td->o.odirect, td->o.sync_io);
-	/* map both O_SYNC / DSYNC to not using NODRAIN */
+	dprint(FD_IO, "td->o.odirect %d td->o.sync_io %d\n",
+					td->o.odirect, td->o.sync_io);
```
Comment (sitsofe) [inline engines/libpmem.c:179]: I think the recent style in fio has been tending towards lining up with the brackets:
```
printf("foo = %d\n",
          i);
```

File: examples/libpmem.fio [hunk 2]
```diff
@@ -17,16 +17,6 @@ thread
...
-#
-# In case of 'scramble_buffers=1', the source buffer
-# is rewritten with a random value every write operations.
-#
-# But when 'scramble_buffers=0' is set, the source buffer isn't
-# rewritten. So it will be likely that the source buffer is in CPU
-# cache and it seems to be high performance.
-#
-scramble_buffers=0
-
```
Comment (sitsofe) [inline examples/libpmem.fio:29]: What made you ditch this? Maybe add the reason to the description...

File: engines/libpmem.c [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * libpmem: IO engine that uses PMDK libpmem to read and write data
```
Comment (sitsofe) [inline engines/libpmem.c:2]: I think the style is that there's an empty line above the S-o-b in the commit message.

File: engines/libpmem.c [hunk 4]
```diff
@@ -212,12 +212,6 @@ static int fio_libpmem_close_file(struct thread_data *td, struct fio_file *f)
 	int ret = 0;
 
 	dprint(FD_IO, "DEBUG fio_libpmem_close_file\n");
-	dprint(FD_IO, "td->o.odirect %d\n", td->o.odirect);
-
-	if (!td->o.odirect) {
-		dprint(FD_IO,"pmem_drain\n");
-		pmem_drain();
-	}
```
Comment (sitsofe) [inline engines/libpmem.c:236]: Out of curiosity do know why the original author might have believed it was?

**Review by sitsofe — COMMENTED** [review]
Only minor comments.

**Review by sitsofe — CHANGES_REQUESTED** [review]
@lukaszstolarczuk any comment on the comments (even if they're minor)?

File: engines/libpmem.c (hunk 4 above)
Comment (lukaszstolarczuk) [reply engines/libpmem.c:236]: I'm not sure. In general you want your data to be persistent on the device (on NVDIMM), so maybe it was added just like that. It's not a mistake it is here, it's just seems redundant.
Anyway, I extended the commit msg a little bit, just FYI.

**Review by lukaszstolarczuk — COMMENTED** [review]
Thanks for the review and ping. Sorry for the delay, I just switched my mind to other tasks and it slipped my mind.

**Review by sitsofe — APPROVED** [review]
LGTM!

@axboe can you pull this one?

File: engines/libpmem.c (hunk 4 above)
Comment (ErwanAliasr1) [reply engines/libpmem.c:236]: Shouldn't we have pmem_drain() before the unmap() to ensure everything is persistent ?

Comment (lukaszstolarczuk) [reply engines/libpmem.c:236]: As I stated in the commit message:
- if we set the direct flag to 1, we use "PMEM_F_MEM_NONTEMPORAL", so non-temporal stores, so we don't need to worry about it (we avoid the need to flush processor caches)
- if we set the direct flag to 0, we simply don't care about this (at least in here, in the benchmarking. We rather don't want that in real-life scenarios)

**Pull request merged**